### ISSUE TITLE
swap pkh.len when running o a big endian machine

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1022]
**Reported by clopez on 26 Jul 2012 01:49 UTC**

Hello,

While doing tests with aircrack-ng I discovered that the first test:

```
./aircrack-ng$(EXE) -w $(TEST_DIR)/password.lst -a 2 -e Harkonen -q $(TEST_DIR)/wpa2.eapol.cap | grep 'KEY FOUND! \[ 12345678 \]' 
```

Will not work on a big endian architecture.

If you run:
```
aircrack-ng -w test/password.lst -a 2 -e Harkonen test/wpa2.eapol.cap 
```

You will get only a message saying "Reading packets, please wait... and nothing more, Aircrack-ng will get stuck at this point.

After digging on the source code I found that the problem was on:

```
if ((int)pkh.len - z < st_cur->wpa.eapol_size )
{
   continue;
}
```

  * On a little endian machine, on this particular test, phk.len contains 99000000 and therefore (int)pkh.len is 153

  * On a big endian machine, on this particular test, phk.len contains also 99000000 and therefore (int)phk.len is -1728053248 thus making all after this fail.

**Comment by aircrack-ng** [comment]
**Attachment by clopez on 26 Jul 2012 01:52 UTC**

Uploaded file [fix-bigendianissues.diff](https://github.com/aircrack-ng/trac-attachments/blob/master/1022/20120726-015210_000-fix-bigendianissues.diff)

**Comment by aircrack-ng** [comment]
**Comment by clopez on 26 Jul 2012 01:49 UTC**

The attached patch fixes this issue.

**Comment by aircrack-ng** [comment]
**Modified by clopez on 26 Jul 2012 01:49 UTC**

**Comment by aircrack-ng** [comment]
**Comment by clopez on 26 Jul 2012 01:49 UTC**

what funny.. just realized that CIGAM is the reverse of MAGIC :D so to check if we are running on a big or little endian machine we just check

```
if (pfh.magic == TCPDUMP_CIGAM)
  printf("im bigendian and file was created on littleendian");
else if (pfh.magic == TCPDUMP_MAGIC)
  printf("im littlendian and file was created on littleendian");
```

is this true or I missed something?

What happens when file was created on bigendian? How this affects #1008?

**Comment by aircrack-ng** [comment]
**Comment by C on 26 Jul 2012 01:49 UTC**

Confirmed. With this patch applied, aircrack-ng compiled on MacOS 10.7 (x86) is able to read big-endian .pcap files (from powerpc32). Thanks for the patch!

**Comment by aircrack-ng** [comment]
**Comment by clopez on 26 Jul 2012 01:49 UTC**

Fixed on aircrack-ng/aircrack-ng@2a0a6b32fdf26aea995c44f31e9abd5789c4f798
